subsys/bluetooth/mesh/adv.c

```diff
@@ -159,22 +176,21 @@ struct net_buf *bt_mesh_adv_buf_get(k_timeout_t timeout)
 	return process_events(events, ARRAY_SIZE(events));
 }
 
-static struct net_buf *bt_mesh_adv_buf_relay_get(k_timeout_t timeout)
-{
-	return net_buf_get(&bt_mesh_relay_queue, timeout);
-}
-
 struct net_buf *bt_mesh_adv_buf_get_by_tag(uint8_t tag, k_timeout_t timeout)
 {
-	if (tag & BT_MESH_LOCAL_ADV) {
-		return bt_mesh_adv_buf_get(timeout);
-	} else if (tag & BT_MESH_RELAY_ADV) {
-		return bt_mesh_adv_buf_relay_get(timeout);
-	} else {
-		return NULL;
+	if (IS_ENABLED(CONFIG_BT_MESH_ADV_EXT_FRIEND_SEPARATE) && tag & BT_MESH_FRIEND_ADV) {
+		return net_buf_get(&bt_mesh_friend_queue, timeout);
+	}
+
+#if CONFIG_BT_MESH_RELAY_ADV_SETS
```

Comment:
(discussed in person)
As we saw, it seems there is not straight forward way to evaluate integer type configuration parameters like you can with boolean types in the `IS_ENABLED()` macro. While I agree that it would be better to align the style I really do prefer using `IS_ENABLED()` rather that `#ifdef` since I think it is more readable. Is this ok for you?